deploy/common/k8sbased.sh

```diff
@@ -163,6 +163,25 @@ function launch_central {
     elif [[ -x "$(command -v roxctl)" && "$(roxctl version)" == "$MAIN_IMAGE_TAG" ]]; then
       echo "Using $(command -v roxctl) for install due to version match with MAIN_IMAGE_TAG $MAIN_IMAGE_TAG"
       use_docker=0
+    else
+      echo "Using docker with $ROXCTL_IMAGE for install. Set USE_LOCAL_ROXCTL=true if you want to use your local version of roxctl"
+      if ! docker pull "$ROXCTL_IMAGE"; then
+        echo "Failed to pull $ROXCTL_IMAGE"
+        exit 1
+      fi
+
+      _check_images() {
+        local images=("$@")
+
+        local image
+        for image in "${images[@]}"; do
+          if ! docker manifest inspect "${image}" > /dev/null; then
+            yes_no_prompt "Couldn't find ${image}. Do you want to continue anyway?" || { echo >&2 "Exiting as requested"; exit 1; }
```

Comment:
Is this something we do elsewhere in the deploy scripts? I haven't personally run it to it ever in my limited use of `deploy-local`.

I'm just thinking of fire-and-forget usage of this script - is it better to block completion here when something goes wrong or log it and continue? Failing to pull roxctl above immediately exits. Is this fatal enough of an error to do the same or would users typically want to continue with the install regardless of problems that will arise?

Reply:
> Is this something we do elsewhere in the deploy scripts?

Yes, but now that I'm looking more carefully at it, we at least check if it's running in CI first: https://github.com/stackrox/stackrox/blob/dff50120e444170448a2f8cab11d6b6f677be790/deploy/common/k8sbased.sh#L150

> is it better to block completion here when something goes wrong or log it and continue?

That's a good question - this thought crossed my mind as well. Honestly, since you use this script more than me, I'm down to follow your lead here. I'm going to add that CI check I mentioned above, but as far as locally using the script, I'm down with implementing whatever is most convenient/expected for the users of the script.

> Failing to pull roxctl above immediately exits. Is this fatal enough of an error to do the same or would users typically want to continue with the install regardless of problems that will arise?

I'm not sure. Echoing what I mentioned above, I'm down to implement whatever makes the most sense to you. The situation I want to avoid is what you brought up in Slack, i.e., it's not clear to a user of the script that something has gone wrong or something is in a state such that ACS won't deploy as expected. I decided to put the prompt in the first iteration to make it *very* obvious that we're currently in that sort of situation. 

Though, now that I've typed all this out, I had the thought of checking if the shell the script is running in is an interactive shell (https://serverfault.com/questions/146745/how-can-i-check-in-bash-if-a-shell-is-running-in-interactive-mode). Maybe a decent compromise is "if in interactive shell, give prompt; else, log a warning."

Anyway, let me know what you'd prefer 🙂

Reply:
For me, either really is fine. Blocking with a prompt might be preferred, just since I run this in an open shell and will notice that the script has stalled when checking to see the progress on my installation. That is probably more noticeable than having the script complete, and then debugging why e.g. scanning isn't working at a later time.

My main concern was how other people might use these scripts, where blocking execution with a prompt might cause issues. IMO though the compromise "if in interactive shell, give prompt; else, log a warning." addresses this and won't cause people running the script headless to wonder why it never completes. 👍